modules/tensor_mechanics/include/actions/CohesiveZoneMasterAction.h

```diff
@@ -17,5 +17,33 @@ class CohesiveZoneMasterAction : public Action
   static InputParameters validParams();
   CohesiveZoneMasterAction(const InputParameters & params);
 
+  /// Method adding the proper relationship manager
+  using Action::addRelationshipManagers;
+  virtual void addRelationshipManagers(Moose::RelationshipManagerType input_rm_type) override;
+
   void act() override;
+
+protected:
+  /// the disaplcements varaible names
+  std::vector<VariableName> _displacements;
+
+  /// number of displacement components
+  const unsigned int _ndisp;
+
+  /// Base name of the material system
+  const std::string _base_name;
+
+  /// strain formulation
+  enum class Kinematic
+  {
+    SmallStrain,
+    TotalLagrangian
```

Comment:
Shouldn't it be `SmallStrain` v.s. `FiniteStrain` or `Incremental` v.s. `Total`?

Reply:
This is really up to you. The input parameter name is `kinematic`, not `strain` so as it is now it make sense, at least for me this nomenclature.
If you prefer strain, let me know but then, in case someone wants to add an updated lagrangian formulation, we will need a `kinematic` parameter distinguishing between the two formulations. Just let me know

Reply:
@jiangwen84 can you tell me your thoughts about this? should I change the parameter `kinematic`->`strain` and use `Finite Strain`

Reply:
To make it consistent with TM, I prefer use Small and Finite as options. What do you think?

Reply:
Fair enough, I'll make the change